src/App.tsx

```diff
@@ -15,36 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SortAlphabetically = 'Sort alphabetically',
+  SortByLength = 'Sort by length',
+  Reverse = 'Reverse',
+  Reset = 'Reset',
+}
+
+const SORT_FIELD_BY_NAME = 'name';
+const SORT_FIELD_BY_LENGTH = 'length';
+
+function getSortedGoods(
+  goods: string[],
+  sortField: string,
+  isReversed: boolean,
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_BY_LENGTH:
+          return good1.length - good2.length;
+        case SORT_FIELD_BY_NAME:
+          return good1.localeCompare(good2);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getSortedGoods(goodsFromServer, sortField, isReversed);
+
+  const resetStates = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  const reverseState = () => {
+    setIsReversed(!isReversed);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
+        <button
+          onClick={() => setSortField(SORT_FIELD_BY_NAME)}
```

Comment:
suggest to make one function factory and use for each sort button.
```
const handleSort = (sortBy: SortBy) => () => setSortField(sortBy)
```

```suggestion
          onClick={handleSort(SORT_FIELD_BY_NAME)}
```